### PR TITLE
[PotentialFlow] Bugfix embedded compressible element and adding tests

### PR DESCRIPTION
**Pull request by marcnunezc**
**Description**
This PR fixes the LHS computation and adds tests with the new finite diff utility.

**Changelog**
- Compute correct LHS
- Add f.d. compressible and incompressible tests.
- Add clamping check for LHS.

### CODE REVIEW

File: applications/CompressiblePotentialFlowApplication/tests/cpp_tests/test_incompressible_potential_element.cpp [hunk 1]
```diff
@@ -209,6 +210,34 @@ namespace Kratos {
...
+      Matrix LHS_finite_diference = ZeroMatrix(number_of_nodes, number_of_nodes);
+      Matrix LHS_analytical = ZeroMatrix(number_of_nodes, number_of_nodes);
+
+      PotentialFlowTestUtilities::ComputeElementalSensitivities<3>(model_part, LHS_finite_diference, LHS_analytical, potential);
+
+      for (unsigned int i = 0; i < LHS_finite_diference.size1(); i++) {
+          for (unsigned int j = 0; j < LHS_finite_diference.size2(); j++) {
+              KRATOS_CHECK_NEAR(LHS_finite_diference(i,j), LHS_analytical(i,j), 1e-10);
+          }
+      }
```
Comment (philbucher) [inline applications/CompressiblePotentialFlowApplication/tests/cpp_tests/test_incompressible_potential_element.cpp]: you can use the matrix compare macro

Comment (marcnunezc) [reply applications/CompressiblePotentialFlowApplication/tests/cpp_tests/test_incompressible_potential_element.cpp]: Ok, I took the chance to update all tests with this

File: applications/CompressiblePotentialFlowApplication/custom_elements/embedded_compressible_potential_flow_element.cpp [hunk 2]
```diff
@@ -100,22 +100,27 @@ void EmbeddedCompressiblePotentialFlowElement<Dim, NumNodes>::CalculateEmbeddedL
...
+        BoundedMatrix<double, NumNodes, NumNodes> aux_matrix = positive_side_weights(i_gauss) * density * prod(DN_DX, trans(DN_DX));
+        noalias(laplacian) += aux_matrix;
+        noalias(rLeftHandSideMatrix) += aux_matrix;
+        if (local_velocity_squared < max_velocity_squared){
+            noalias(rLeftHandSideMatrix) += positive_side_weights(i_gauss) * 2 * DrhoDu2 * outer_prod(DNV, trans(DNV));
+        }
     }
 
-    noalias(rRightHandSideVector) = -prod(rLaplacianMatrix, potential);
+    noalias(rRightHandSideVector) = -prod(laplacian, potential);
```
Comment (inigolcanalejo) [inline applications/CompressiblePotentialFlowApplication/custom_elements/embedded_compressible_potential_flow_element.cpp:123]: IMO it is better style to compute RHS and LHS separately, but it just might be a matter of taste

**Review by inigolcanalejo — APPROVED** [review]
Looks good to me, thanks for updating the tests!

File: applications/CompressiblePotentialFlowApplication/custom_elements/embedded_compressible_potential_flow_element.cpp (hunk 2 above)
Comment (marcnunezc) [reply applications/CompressiblePotentialFlowApplication/custom_elements/embedded_compressible_potential_flow_element.cpp:123]: True, my only issue with that is that I cannot use the same for-loop to compute both at the same time.

**Pull request merged**